### PR TITLE
#106 Flat message hierarchy

### PR DESCRIPTION
**Pull request by jaytaph**

### CODE REVIEW

File: cmd/bm-server/internal/account/file_message.go [hunk 1]
```diff
@@ -138,22 +142,42 @@ func (r *fileRepo) MoveToBox(addr hash.Hash, srcBox, dstBox int, msgID string) e
...
-	// if box == "inbox" {
-	// 	dstPath = r.getPath(addr, filepath.Join(box, fmt.Sprintf("%d-%s", time.Now().Unix(), msgID)))
-	// }
+	// Finally, move the message to our message directory
+	dstPath := r.getPath(addr, filepath.Join(messageDir, msgID))
 	return r.fs.Rename(srcPath, dstPath)
 }
+
+// RemoveMessage Removes a message complete from the account
+func (r *fileRepo) RemoveMessage(addr hash.Hash, msgID string) error {
```
Comment (acalatrava) [inline cmd/bm-server/internal/account/file_message.go:159]: When removing a message we should check all their boxes to remove the symlinks too

Comment (acalatrava) [reply cmd/bm-server/internal/account/file_message.go:159]: Probably the easiest way is to enumerate all the boxes and call RemoveFromBox

**Review by acalatrava — APPROVED** [review]
LGTM but we should fix the RemoveMessage before merging

File: cmd/bm-server/internal/account/file_message.go (hunk 1 above)
Comment (jaytaph) [reply cmd/bm-server/internal/account/file_message.go:159]: True. Will fix

**Comment by acalatrava** [comment]
We should also change the api
```
	authRouter.HandleFunc("/account/{addr:[A-Za-z0-9]{64}}/box/{box:[0-9]+}/message/{message:[A-Za-z0-9-]+}", handler.GetMessage).Methods("GET")
	authRouter.HandleFunc("/account/{addr:[A-Za-z0-9]{64}}/box/{box:[0-9]+}/message/{message:[A-Za-z0-9-]+}/block/{block:[A-Za-z0-9-]+}", handler.GetMessageBlock).Methods("GET")
	authRouter.HandleFunc("/account/{addr:[A-Za-z0-9]{64}}/box/{box:[0-9]+}/message/{message:[A-Za-z0-9-]+}/attachment/{attachment:[A-Za-z0-9-]+}", handler.GetMessageAttachment).Methods("GET")
```

**Review by acalatrava — APPROVED** [review]
LGTM!

**Comment by jaytaph** [comment]
Fixed #106

**Pull request merged**